_overviews/scala3-book/taste-contextual-abstractions.md

```diff
@@ -12,19 +12,27 @@ next-page: taste-toplevel-definitions
 TODO: Now that this is a separate section, it needs a little more content.
 {% endcomment %}
 
+Under certain circumstances, you can omit some parameters of method calls that are considered repetitive.
 
-Under certain circumstances, the Scala compiler can “write” some parts of your programs.
-For instance, consider a program that sorts a list of addresses by two criteria, city name and then street name:
+Those parameters are called _Context Parameters_ because there are inferred by the compiler from the context surrounding the method call.
+
+For instance, consider a program that sorts a list of addresses by two criteria: city name and then street name.
 
 ```scala
 val addresses: List[Address] = ...
-addresses.sortBy(address => (address.city, address.street))
+
+given ordering: Ordering[Address] = Ordering.by(address => (address.city, address.street))
+
+addresses.sorted(using ordering)
+addresses.sorted
 ```
 
-The sorting algorithm needs to compare addresses by first comparing their city names, and then also their street names when the city names are the same.
-However, with the use of contextual abstraction, you don’t need to manually define this ordering relation, because the compiler is able to summon it automatically based on an existing ordering relation for comparing string values.
+The ordering logic is defined by the `ordering` instance wich is made available in the current scope by the `given` keyword.
+The last two lines, that sort the list of addresses, are strictly similar.
```

Comment:
```suggestion
The last two lines, that sort the list of addresses, are equivalent.
```
?

Reply:
I pushed `strictly equivalent`.
I want to keep the `strictly` to stress the fact that there is nothing that differs in the produced byte-code.